app/models/amazon_agent_manager_worker/runner.rb

```diff
@@ -0,0 +1,91 @@
+require 'amazon_ssa_support'
+
+class AmazonAgentManagerWorker::Runner < MiqWorker::Runner
+  def do_before_work_loop
+    @managers = self.class.all_amazon_agent_manager_in_zone
```

Comment:
What happens if the worker bounces?  You'll have a bunch of running instances up in Amazon, but those are not reflected here.  Does there need to be some kind of initialize synchronization?

Reply:
If the worker bounces just during the deploying state, like what you said, the new worker will launch another agent. We will have a bunch of instances in Amazon. We discussed this before and have the conclusion:
1. This situation is a corner case, it not always happens;
2. Even multiple agents are launched, if there is no requests coming in, they will power themselves off in 15.minutes;
3. Eventually we will add a cleanup logic to handle all these leftovers.